ui/src/app/pages/analysis/interactive-notebook.tsx

```diff
@@ -149,28 +149,39 @@ export const InteractiveNotebook = fp.flow(
 
     componentDidMount(): void {
       this.props.hideSpinner();
-    }
 
-    componentDidUpdate(prevProps: Readonly<Props>) {
       const {urlParams: {ns, wsid, nbName}} = this.props;
+      if (ns && wsid && nbName) {
+        this.loadNotebook();
+      }
+    }
 
-      if (!hasNewValidProps(this.props, prevProps, [p => p.urlParams.ns, p => p.urlParams.wsid, p => p.urlParams.nbName])) {
```

Comment:
We weren't loading this on mount (because sometimes the params haven't loaded yet) and React batches similar changes so it rerenders less so this conditional was always passing